File: DiscordNitroGen.java
import java.security.SecureRandom;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class DiscordNitroGen {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Ingrese la cantidad de promoCodes que desea generar: ");
        int cantidadPromoCodes = scanner.nextInt();

        int longitudNumeros = 19;
        int longitudAlfanumerico1 = 39;
        int longitudComplejo = 238; // 5 guiones y 233 caracteres alfanuméricos
        int longitudAlfanumerico2 = 16;

        List<String> PromoCodes = generarPromocodes(cantidadPromoCodes, longitudNumeros, longitudAlfanumerico1, longitudComplejo, longitudAlfanumerico2);

        System.out.println("PromoCodes generados:");
        for (String Promocodes : PromoCodes) {
            System.out.println(PromoCodes);
        }

        scanner.close();
    }

    public static List<String> generarPromoCodes(int cantidad, int longitudNumeros, int longitudAlfanumerico1, int longitudComplejo, int longitudAlfanumerico2) {
        List<String> licencias = new ArrayList<>();
        for (int i = 0; i < cantidad; i++) {
            String numeros = generarVariableNumerica(longitudNumeros);
            String alfanumerico1 = generarVariableAlfanumerica(longitudAlfanumerico1);
            String complejo = generarVariableCompleja(longitudComplejo);
            String alfanumerico2 = generarVariableAlfanumerica(longitudAlfanumerico2);
            licencias.add("https://discord.com/billing/partner-promotions/" + numeros + "/" + alfanumerico1 + ".." + alfanumerico2 + "." + complejo);
        }
        return PromoCodes;
    }

    private static String generarVariableNumerica(int longitud) {
        SecureRandom random = new SecureRandom();
        StringBuilder sb = new StringBuilder(longitud);
        String numeros = "0123456789";

        for (int i = 0; i < longitud; i++) {
            int index = random.nextInt(numeros.length());
            sb.append(numeros.charAt(index));
        }
        return sb.toString();
    }

    private static String generarVariableAlfanumerica(int longitud) {
        SecureRandom random = new SecureRandom();
        StringBuilder sb = new StringBuilder(longitud);
        String caracteres = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";

        for (int i = 0; i < longitud; i++) {
            int index = random.nextInt(caracteres.length());
            sb.append(caracteres.charAt(index));
        }
        return sb.toString();
    }

    private static String generarVariableCompleja(int longitud) {
        SecureRandom random = new SecureRandom();
        StringBuilder sb = new StringBuilder(longitud);
        String caracteres = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";

        // Posiciones aleatorias para colocar guiones
        List<Integer> posiciones = new ArrayList<>();
        for (int i = 0; i < 5; i++) {
            posiciones.add(i * (longitud / 5) + random.nextInt(longitud / 5)); // Distribuir guiones en diferentes partes
        }
        Collections.shuffle(posiciones); // Mezclar las posiciones para colocar guiones en diferentes lugares

        int guionIndex = 0;
        for (int i = 0; i < longitud; i++) {
            if (guionIndex < 5 && posiciones.contains(i)) {
                sb.append('-');
                guionIndex++;
            } else {
                int index = random.nextInt(caracteres.length());
                sb.append(caracteres.charAt(index));
            }
        }
        return sb.toString();
    }
}

File: DiscordNitroGen.py
import random

def generar_variable_numerica(longitud):
    numeros = "0123456789"
    return ''.join(random.choice(numeros) for _ in range(longitud))

def generar_variable_alfanumerica(longitud):
    caracteres = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
    return ''.join(random.choice(caracteres) for _ in range(longitud))

def generar_variable_compleja(longitud):
    caracteres = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"

    # Posiciones aleatorias para colocar guiones
    posiciones = sorted(random.sample(range(longitud), 5))

    guion_index = 0
    complejo = ''
    for i in range(longitud):
        if guion_index < 5 and posiciones[guion_index] == i:
            complejo += '-'
            guion_index += 1
        else:
            complejo += random.choice(caracteres)
    return complejo

def generar_promocodes(cantidad, longitud_numeros, longitud_alfanumerico1, longitud_complejo, longitud_alfanumerico2):
    promocodes = []
    for _ in range(cantidad):
        numeros = generar_variable_numerica(longitud_numeros)
        alfanumerico1 = generar_variable_alfanumerica(longitud_alfanumerico1)
        complejo = generar_variable_compleja(longitud_complejo)
        alfanumerico2 = generar_variable_alfanumerica(longitud_alfanumerico2)
        promocodes.append(f"https://discord.com/billing/partner-promotions/{numeros}/{alfanumerico1}..{alfanumerico2}.{complejo}")
    return promocodes

if __name__ == "__main__":
    cantidad_promocodes = int(input("Ingrese la cantidad de promoCodes que desea generar: "))
    longitud_numeros = 19
    longitud_alfanumerico1 = 39
    longitud_complejo = 238  # 5 guiones y 233 caracteres alfanuméricos
    longitud_alfanumerico2 = 16

    promo_codes = generar_promocodes(cantidad_promocodes, longitud_numeros, longitud_alfanumerico1, longitud_complejo, longitud_alfanumerico2)

    print("PromoCodes generados:")
    for promo_code in promo_codes:
        print(promo_code)

    input("Presione Enter para salir...")
